paths_cli/tests/compiling/test_core.py

```diff
@@ -0,0 +1,319 @@
+        # all_objs list and leave the named_objs dict unchanged
+        obj = MockNamedObject('foo')
+        assert obj.name is None
+        assert self.compiler.all_objs == []
+        assert self.compiler.named_objs == {}
+        obj = self.compiler.register_object(obj, None)
+        assert obj.name is None
+        assert self.compiler.all_objs == [obj]
+        assert self.compiler.named_objs == {}
+
+    def test_register_object_duplicate(self):
+        # if an attempt is made to register an object with a name that is
+        # already in use, an InputError should be raised, and the object
+        # should not register with either all_objs or named_objs
+        obj = MockNamedObject('foo').named('bar')
+        self.compiler.named_objs['bar'] = obj
+        self.compiler.all_objs.append(obj)
+        obj2 = MockNamedObject('baz')
+        with pytest.raises(InputError, match="already exists"):
+            self.compiler.register_object(obj2, 'bar')
+
+        assert self.compiler.named_objs == {'bar': obj}
+        assert self.compiler.all_objs == [obj]
+        assert obj2.name is None
+
+    def test_register_builder(self):
+        # a new builder can be registered and used, if it has a new name
+        assert len(self.compiler.type_dispatch) == 1
+        assert 'bar' not in self.compiler.type_dispatch
+        self.compiler.register_builder(lambda dct: 10, 'bar')
+        assert len(self.compiler.type_dispatch) == 2
+        assert 'bar' in self.compiler.type_dispatch
+        input_dict = {'type': 'bar'}
+        assert self.compiler(input_dict) == 10
+
+    def test_register_builder_duplicate(self):
+        # if an attempt is made to registere a builder with a name that is
+        # already in use, a RuntimeError is raised
+        orig = self.compiler.type_dispatch['foo']
+        with pytest.raises(RuntimeError, match="already registered"):
+            self.compiler.register_builder(lambda dct: 10, 'foo')
+
+        assert self.compiler.type_dispatch['foo'] is orig
+
+    def test_register_builder_identical(self):
+        # if an attempt is made to register a builder that has already been
+        # registered, nothin happens (since it is already registered)
+        orig = self.compiler.type_dispatch['foo']
+        self.compiler.register_builder(orig, 'foo')
+
+    @pytest.mark.parametrize('input_type', ['str', 'dict'])
+    def test_compile(self, input_type):
+        # the compile method should work whether the input is a dict
+        # representing an object to be compiled or string name for an
+        # already-compiled object
+        pytest.skip()
+
+    @pytest.mark.parametrize('input_type', ['str', 'dict'])
+    @pytest.mark.parametrize('as_list', [True, False])
+    def test_call(self, input_type, as_list):
+        # the call method should work whether the input is a single object
+        # or a list of objects (as well as whether string or dict)
+        pytest.skip()
```

Comment:
still needs to be implemented or left for other PRs? (if left, please add a TODO in the `skip()` call)

Reply:
Implemented skipped tests